src/compiler/scanner.ts

```diff
@@ -983,10 +985,17 @@ namespace ts {
             else {
                 tokenValue = result;
                 const type = checkBigIntSuffix(); // if value is an integer, check whether it is a bigint
+                checkForIdentifierStartAfterNumericLiteral();
                 return { type, value: tokenValue };
             }
         }
 
+        function checkForIdentifierStartAfterNumericLiteral() {
+            if (isIdentifierStart(text.charCodeAt(pos), languageVersion)) {
+                error(Diagnostics.An_identifier_cannot_follow_a_numeric_literal, pos, 1);

```

Comment:
Question for @DanielRosenwasser. Should we report error on whole identifier instead of single character?

Reply:
That would be nicer; do you think you'd be able to integrate that in @JoshuaKGoldberg?

Reply:
Extended to the full identifier - note that this now means we have to scan the identifier to capture its length _(right?)_. I reset the length after to not affect parsing the identifier otherwise.